homeassistant/components/flux_led/light.py

```diff
@@ -428,7 +433,16 @@ async def async_turn_on(self, **kwargs):
             return
 
         if effect in EFFECT_MAP:
+            self._current_effect = effect
+            self.async_write_ha_state()
```

Comment:
Why did we change to not end with the state update?

Reply:
Because the call to the light sometimes takes a few seconds by calling in this order it avoids the UI from flipping the state/effect back and forth if the update triggers while the call to the light is happening.